vmm/src/memory_manager.rs

```diff
@@ -1368,6 +1404,67 @@ impl MemoryManager {
     pub fn memory_zones(&self) -> &MemoryZones {
         &self.memory_zones
     }
+
+    pub fn dirty_memory_range_table(
+        &self,
+    ) -> std::result::Result<MemoryRangeTable, MigratableError> {
```

Comment:
Maybe we should also return the `total_pages` to be able to start measuring our migration convergence.

Reply:
I thought about this originally but I think the number of ranges is a more useful metric to gauge convergence as that is the granularity we are transferring on.